modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -2086,9 +2093,13 @@ public UpdateResult executeUpdateOnDataNode(
 
         GridFutureAdapter<Void> rebuildCacheIdxFut = new GridFutureAdapter<>();
 
-        //to avoid possible data race
+        // To avoid possible data race.
         GridFutureAdapter<Void> outRebuildCacheIdxFut = new GridFutureAdapter<>();
 
+        // An internal feature for the ability to cancel index rebuilding.
+        SchemaIndexCacheFuture intlRebFut = new SchemaIndexCacheFuture(new SchemaIndexOperationCancellationToken());
+        cancelIndexRebuildFuture(idxRebuildFuts.put(cctx.cacheId(), intlRebFut));
```

Comment:
we don`t need to give a chance to run sequentially rebuild the same indexes, as i understand ..